compiler/src/lsp/dune

```diff
@@ -0,0 +1,8 @@
+(library
+ (name lsp)
+ (public_name lsp)
+ (synopsis "LSP modules for Grain compiler")
+ (libraries compiler-libs.common dyp grain_utils grain_typed utf8.lib str
+   ppx_deriving_yojson.runtime ppx_sexp_conv.runtime-lib sexplib yojson)
```

Comment:
I think we need to trim this down. I don't believe all these will be be needed for this package (especially `dyp`).

Reply:
Good point, I was lazy and just copied the existing project.  I'll trim it down to what we need

Reply:
Trimmed down.